src/psyclone/psyir/transformations/inline_trans.py

```diff
@@ -515,45 +524,81 @@ def _find_routine(call_node):
             for routine in call_node.root.children:
                 if isinstance(routine, Routine) and routine.name == name:
                     return routine
+            else:
+                raise TransformationError(
+                    f"Failed to find the source code of the unresolved "
+                    f"routine '{routine.name}' after trying wildcard imports "
+                    f"'{wildcard_names}' and all routines that are not in "
+                    f"containers.")
 
         elif routine_sym.is_import:
             container_symbol = routine_sym.interface.container_symbol
             routine = InlineTrans._find_routine_in_container(
                 call_node, container_symbol)
             if routine:
                 return routine
-
-        raise TransformationError(
-            f"Failed to find the source for routine '{name}' and "
-            f"therefore cannot inline it.")
+            raise TransformationError(
+                f"Failed to find the source for the imported routine "
+                f"'{routine_sym.name}' and therefore cannot inline it.")
+        else:
+            raise InternalError(
+                f"Routine Symbol '{routine_sym.name}' is not local, "
+                f"unresolved or imported.")
 
     @staticmethod
-    def _find_routine_in_container(
-            call_node, container_symbol):
-        ''' xxx '''
+    def _find_routine_in_container(call_node, container_symbol):
+        '''Searches for the definition of a routine that is being called by
+        the supplied Call. This routine must exist within a container
```

Comment:
Please extend to make it clear that if the routine isn't found then `None` is returned. e.g. "...Call. If present, this routine must..."

Reply:
Better, thanks.